### ISSUE TITLE
Why is `cpr::Body` derived from `std::string`?

### ISSUE THREAD

**Issue by cyyzero** [whoshuu/cpr#303]
https://github.com/whoshuu/cpr/blob/3d14e61ed247a90a07db26fb107beb3086a035d6/include/cpr/body.h#L13

Is there a risk of memory leak? Because `std::string` has no virtual destructor.

**Comment by COM8** [comment]
Valid argument.
And it's definitely a problem.
And `Body` does not introduce any new methods.
So we could replace `Body` with something like `typedef std::string Body;`
**But** we already have `typedef std::string Url;`
If we now want to do something like this:
```C++
void SetOption(const Body& body);
void SetOption(const Url& url);
```
This would fail since both are just:
```C++
void SetOption(const std::string& body);
void SetOption(const std::string& url);
```

### The solution:
Replace `Body` and `Url` with simple wrapper classes or structs:
```C++
struct Body {
  Body(std::string str) : _str(str) {}
  std::string& str() {return _str;}
  private:
  std::string _str;
}
struct Url {
  Body(std::string str) : _str(str) {}
  std::string& str() {return _str;}
  private:
  std::string _str;
}
```

This would solve the issue. To be actually usable, this needs more ways to construct the `Body` and `Url` objects.

Since this is a breaking change, we could introduce this perhaps in version 4.0 of CPR.

**Comment by COM8** [comment]
Fixed in c77dbac12712af56d207637745691d6b4c3d0308
